rclpy/src/rclpy/qos.cpp

```diff
@@ -131,11 +128,7 @@ create_qos_profile(
 py::dict
 convert_qos_profile_to_dict(const rmw_qos_profile_t & qos_profile)
 {
-  PyObject * pydict = rclpy_common_convert_to_qos_dict(&qos_profile);
-  if (!pydict) {
-    throw py::error_already_set();
-  }
-  return py::reinterpret_steal<py::dict>(pydict);
+  return convert_to_qos_dict(&qos_profile);
```

Comment:
Since this function `convert_qos_profile_to_dict` is just passing through to `convert_to_qos_dict`, I think you could delete this function and make the `def()` below

```
.def("to_dict", &convert_to_qos_dict)
```